### PR TITLE
Virtual table to list all the hard limits.

### PR DESCRIPTION
**Pull request by nirbhayc**

### CODE REVIEW

File: csc2/macc.h [hunk 1]
```diff
@@ -75,7 +75,7 @@ extern char *maccfuncpath;
 
 /* MAXINDEX HAS BECOME OBSOLETE - FOR NOW, JUST KLUDGE TO FIX */
 #define MAXKEYS 256  /* maximum # of keys with cases */
-#define MAXINDEX 256 /* max # of indices */
+//#define MAXINDEX 256 /* max # of indices */
```
Comment (nirbhayc) [inline csc2/macc.h]: Do we really need this here? For now, I have commented it in favor of one defined in bbinc/cdb2_constants.h (the value of which is 50 though).

File: sqlite/ext/comdb2/limits.c [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        #LIMIT, DESCR, LIMIT                                                   \
+    }
+
+struct limit_t {
+    const char *name;
+    const char *descr;
+    int limit;
+} limits[] = {
+    /* COMDB2_LIMIT(COMDB2_MAX_RECORD_SIZE, "????"), */
+    /* COMDB2_LIMIT(LONG_REQMS, "????"), */
```
Comment (nirbhayc) [inline sqlite/ext/comdb2/limits.c]: Is this a good candidate to be listed in this table?

File: sqlite/ext/comdb2/limits.c [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+    COMDB2_LIMIT(MAXBLOBLENGTH, "Maximum blob length"),
+    COMDB2_LIMIT(MAXBLOBS, "Maximum number of blobs"),
+    COMDB2_LIMIT(MAXCOLNAME, "Maximum column name length"),
+    COMDB2_LIMIT(MAXCOLUMNS, "Maximum columns in a table"),
+    COMDB2_LIMIT(MAXCONSTRAINTS, "Maximum number of constraint in a table"),
+    COMDB2_LIMIT(MAXCONSUMERS, "Maximum queue consumers"),
+    COMDB2_LIMIT(MAXCUSTOPNAME, "Maximum length of a custom operation name"),
+    COMDB2_LIMIT(MAX_DBNAME_LENGTH, "Maximum length of database name"),
+    COMDB2_LIMIT(MAXDTASTRIPE, "Maximum number of data stripes"),
+    /* COMDB2_LIMIT(MAXDYNTAGCOLUMNS, "????"), */
```
Comment (nirbhayc) [inline sqlite/ext/comdb2/limits.c]: What's this limit for? I couldn't derive a good description for this from the code.

File: sqlite/ext/comdb2/limits.c [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+    {                                                                          \
+        #LIMIT, DESCR, LIMIT                                                   \
+    }
+
+struct limit_t {
+    const char *name;
+    const char *descr;
+    int limit;
+} limits[] = {
+    /* COMDB2_LIMIT(COMDB2_MAX_RECORD_SIZE, "????"), */
```
Comment (nirbhayc) [inline sqlite/ext/comdb2/limits.c]: Is this same as MAXLRL?

File: bbinc/cdb2_constants.h [hunk 5]
```diff
@@ -17,48 +17,31 @@
...
-    ,
-    MAXPSTRLEN = 256,
-    MAXTABLELEN = 32,
-    MAXCOMDBGNODES = 32,
-    MAXBLOBLENGTH = 256 * 1024 * 1024 /* TODO set a good maximum here */
-    ,
-    MAXCONSTRAINTS = 32,
-    MAXCONSUMERS = 32 /* max queue consumers, to match bdblib limit */
-    ,
-    MAXCUSTOPNAME = 32 /* max length of a custom operation name */
```
Comment (nirbhayc) [inline bbinc/cdb2_constants.h]: This one's no longer used anywhere. Gone.

File: bbinc/cdb2_constants.h [hunk 6]
```diff
@@ -17,48 +17,31 @@
...
-    MAXSIBLINGS = 64,
-    MAXCOLUMNS = 1024 /*max columns in table, hard limit*/
-    ,
-    MAXDYNTAGCOLUMNS = 2048,
-    MAXTAGLEN = 64,
-    MAXCOLNAME = 99 /*maximum column name length (not incl. \0)*/
-    ,
-    MAXPSTRLEN = 256,
-    MAXTABLELEN = 32,
-    MAXCOMDBGNODES = 32,
```
Comment (nirbhayc) [inline bbinc/cdb2_constants.h]: This one's no longer used anywhere. Gone.

File: bbinc/cdb2_constants.h [hunk 7]
```diff
@@ -17,48 +17,31 @@
...
-    MAXCONSTRAINTS = 32,
-    MAXCONSUMERS = 32 /* max queue consumers, to match bdblib limit */
-    ,
-    MAXCUSTOPNAME = 32 /* max length of a custom operation name */
-    ,
-    MAXDTASTRIPE = 16,
-    MAX_QUEUE_HITS_PER_TRANS = 8 /* how many queue hits we efficiently
-                                    remember per trans */
-    ,
-    MAXALIASNAME = 32,
```
Comment (nirbhayc) [inline bbinc/cdb2_constants.h]: This one's no longer used anywhere. Gone.

File: sqlite/ext/comdb2/limits.c [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+
+#if defined(SQLITE_BUILDING_FOR_COMDB2) && !defined(SQLITE_CORE)
+#define SQLITE_CORE 1
+#endif
+
+#include <assert.h>
+#include "comdb2systbl.h"
+#include "comdb2systblInt.h"
+#include "cdb2_constants.h"
+
```
Comment (ldennis) [inline sqlite/ext/comdb2/limits.c]: Again, this seems to break build with internal builddb in bbenv as I mentioned in #212. Consider adding `#include <string.h>`?

File: sqlite/ext/comdb2/limits.c (hunk 2 above)
Comment (mponomar) [inline sqlite/ext/comdb2/limits.c]: COMDB2_MAX_RECORD_SIZE  = 16k
LONG_REQMS isn't a limit, it's just an alarming threshold for requests.

File: sqlite/ext/comdb2/limits.c (hunk 3 above)
Comment (mponomar) [inline sqlite/ext/comdb2/limits.c]: MAXDYNTAGCOLUMNS - effectively the max number of bound parameter values per statement.

File: sqlite/ext/comdb2/limits.c (hunk 8 above)
Comment (nirbhayc) [inline sqlite/ext/comdb2/limits.c]: https://github.com/bloomberg/comdb2/pull/232

**Pull request merged**